locals.tf

```diff
@@ -0,0 +1,103 @@
+/**
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#------------------------------#
+# Random string for deployment #
+#------------------------------#
+resource "random_id" "random_hash_suffix" {
+  byte_length = 4
+}
+
+#--------#
+# Locals #
+#--------#
+locals {
+  random_hash             = "${random_id.random_hash_suffix.hex}"
+  root_resource_id        = "${var.org_id != "" ? "organizations/${var.org_id}" : var.folder_id != "" ? "folders/${var.folder_id}" : ""}"
+  vpc_host_project_id     = "${var.vpc_host_project_id}"
+  server_zone             = "${var.server_region}-c"
+  server_startup_script   = "${file("${path.module}/scripts/forseti-server/forseti_server_startup_script.sh.tpl")}"
+  server_environment      = "${file("${path.module}/scripts/forseti-server/forseti_environment.sh.tpl")}"
+  server_env              = "${file("${path.module}/scripts/forseti-server/forseti_env.sh.tpl")}"
+  server_conf             = "${file("${path.module}/configs/forseti_conf_server.yaml")}"
+  server_conf_path        = "${var.forseti_home}/configs/forseti_conf_server.yaml"
+  server_name             = "forseti-server-vm-${local.random_hash}"
+  server_sa_name          = "forseti-server-gcp-${local.random_hash}"
+  cloudsql_name           = "forseti-server-db-${local.random_hash}"
+  storage_bucket_name     = "forseti-server-${local.random_hash}"
+  storage_cai_bucket_name = "forseti-cai-export-${local.random_hash}"
+  server_bucket_name      = "forseti-server-${local.random_hash}"
+
+  client_startup_script = "${file("${path.module}/scripts/forseti-client/forseti_client_startup_script.sh.tpl")}"
+  client_env_script     = "${file("${path.module}/scripts/forseti-client/forseti_environment.sh.tpl")}"
+  client_conf           = "${file("${path.module}/configs/forseti_conf_client.yaml")}"
+  client_conf_path      = "${var.forseti_home}/configs/forseti_conf_client.yaml"
+  client_sa_name        = "forseti-client-gcp-${local.random_hash}"
```

Comment:
Any particular reason for having the `random_hash` on the service account?

Reply:
just following the Deployment Manager implementation here